cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -124,6 +126,17 @@ public GoalOptimizer(KafkaCruiseControlConfig config,
     _proposalGenerationException = new AtomicReference<>();
     _proposalPrecomputingProgress = new OperationProgress();
     _proposalComputationTimer = dropwizardMetricRegistry.timer(MetricRegistry.name(GOAL_OPTIMIZER_SENSOR, "proposal-computation-timer"));
+
+    // The cluster is identified as under-provisioned, over-provisioned, or right-sized (undecided not reported).
+    dropwizardMetricRegistry.register(MetricRegistry.name(GOAL_OPTIMIZER_SENSOR, "under-provisioned"),
```

Comment:
It looks like you're creating new metrics rather than share the same metric with the ones in AnomalyDetectorManager. I think we should use the same metric and emit values from both places.

Reply:
Name is modified in new commit.